File: GKPageScrollView/Demo/网易云音乐/GKWYHeaderView.m
//
//  GKWYHeaderView.m
//  GKPageScrollView
//
//  Created by QuintGao on 2018/10/28.
//  Copyright © 2018 QuintGao. All rights reserved.
//

#import "GKWYHeaderView.h"

@interface GKWYHeaderView()

@property (nonatomic, strong) UILabel               *countLabel;

@property (nonatomic, strong) UILabel               *tagLabel;

@property (nonatomic, strong) UIButton              *personalBtn;

@property (nonatomic, strong) UIButton              *collectBtn;

@end

@implementation GKWYHeaderView

- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = [UIColor clearColor];
        [self addSubview:self.nameLabel];
        [self addSubview:self.countLabel];
        [self addSubview:self.tagLabel];
        [self addSubview:self.personalBtn];
        [self addSubview:self.collectBtn];
        
        // 110 44
        [self.tagLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.bottom.equalTo(self).offset(-ADAPTATIONRATIO * 26.0f);
            make.left.equalTo(self).offset(ADAPTATIONRATIO * 15.0f);
        }];
        
        [self.countLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(self.tagLabel);
            make.bottom.equalTo(self.tagLabel.mas_top).offset(-ADAPTATIONRATIO * 12.0f);
        }];
        
        [self.nameLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.left.equalTo(self.tagLabel);
            make.bottom.equalTo(self.countLabel.mas_top).offset(-20.0f);
        }];
        
        [self.collectBtn mas_makeConstraints:^(MASConstraintMaker *make) {
            make.bottom.equalTo(self.tagLabel.mas_bottom);
            make.right.equalTo(self).offset(-ADAPTATIONRATIO * 20.0f);
            make.width.mas_equalTo(ADAPTATIONRATIO * 150.0f);
            make.height.mas_equalTo(ADAPTATIONRATIO * 50.0f);
        }];
        
        [self.personalBtn mas_makeConstraints:^(MASConstraintMaker *make) {
            make.bottom.equalTo(self.tagLabel.mas_bottom);
            make.right.equalTo(self.collectBtn.mas_left).offset(-ADAPTATIONRATIO * 20.0f);
            make.width.mas_equalTo(ADAPTATIONRATIO * 150.0f);
            make.height.mas_equalTo(ADAPTATIONRATIO * 50.0f);
        }];
    }
    return self;
}

#pragma mark - 懒加载
- (UILabel *)nameLabel {
    if (!_nameLabel) {
        _nameLabel = [UILabel new];
        _nameLabel.font = [UIFont boldSystemFontOfSize:18.0f];
        _nameLabel.textColor = [UIColor whiteColor];
        _nameLabel.text = @"展展与罗罗";
    }
    return _nameLabel;
}

- (UILabel *)countLabel {
    if (!_countLabel) {
        _countLabel = [UILabel new];
        _countLabel.font = [UIFont systemFontOfSize:13.0f];
        _countLabel.textColor = [UIColor whiteColor];
        _countLabel.text = @"被收藏了60182次";
    }
    return _countLabel;
}

- (UILabel *)tagLabel {
    if (!_tagLabel) {
        _tagLabel = [UILabel new];
        _tagLabel.font = [UIFont systemFontOfSize:13.0f];
        _tagLabel.textColor = [UIColor whiteColor];
        _tagLabel.text = @"网页音乐人";
    }
    return _tagLabel;
}

- (UIButton *)personalBtn {
    if (!_personalBtn) {
        _personalBtn = [UIButton new];
        _personalBtn.titleLabel.font = [UIFont systemFontOfSize:14.0f];
        [_personalBtn setTitle:@"个人主页" forState:UIControlStateNormal];
        [_personalBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
        _personalBtn.layer.cornerRadius = ADAPTATIONRATIO * 25.0f;
        _personalBtn.layer.masksToBounds = YES;
        _personalBtn.layer.borderColor = [UIColor whiteColor].CGColor;
        _personalBtn.layer.borderWidth = 0.5f;
    }
    return _personalBtn;
}

- (UIButton *)collectBtn {
    if (!_collectBtn) {
        _collectBtn = [UIButton new];
        _collectBtn.backgroundColor = [UIColor redColor];
        _collectBtn.titleLabel.font = [UIFont systemFontOfSize:14.0f];
        [_collectBtn setTitle:@"收藏" forState:UIControlStateNormal];
        [_collectBtn setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
        _collectBtn.layer.cornerRadius = ADAPTATIONRATIO * 22.0f;
        _collectBtn.layer.masksToBounds = YES;
    }
    return _collectBtn;
}

@end

File: SwiftExample/SwiftExample/Demo/GKPageScrollView/网易云音乐/GKWYHeaderView.swift
//
//  GKWYHeaderView.swift
//  GKPageScrollViewSwift
//
//  Created by gaokun on 2019/2/22.
//  Copyright © 2019 gaokun. All rights reserved.
//

import UIKit
import GKPageScrollView

public let kWYHeaderHeight = (kScreenW * 500.0 / 750.0 - kNavBar_Height)

class GKWYHeaderView: UIView {
    lazy var nameLabel: UILabel! = {
        let nameLabel = UILabel()
        nameLabel.font = UIFont.boldSystemFont(ofSize: 18.0)
        nameLabel.textColor = UIColor.white
        nameLabel.text = "展展与罗罗"
        return nameLabel
    }()
    
    lazy var countLabel: UILabel! = {
        let countLabel = UILabel()
        countLabel.font = UIFont.systemFont(ofSize: 13.0)
        countLabel.textColor = UIColor.white
        countLabel.text = "被收藏了60182次"
        return countLabel
    }()
    
    lazy var tagLabel: UILabel! = {
        let tagLabel = UILabel()
        tagLabel.font = UIFont.systemFont(ofSize: 13.0)
        tagLabel.textColor = UIColor.white
        tagLabel.text = "网易音乐人"
        return tagLabel
    }()
    
    lazy var personalBtn: UIButton! = {
        let btn = UIButton()
        btn.titleLabel?.font = UIFont.systemFont(ofSize: 14.0)
        btn.setTitle("个人主页", for: .normal)
        btn.setTitleColor(UIColor.white, for: .normal)
        btn.layer.cornerRadius = ADAPTATIONRATIO * 25.0
        btn.layer.masksToBounds = true
        btn.layer.borderColor = UIColor.white.cgColor
        btn.layer.borderWidth = 0.5
        return btn
    }()
    
    lazy var collectBtn: UIButton! = {
        let btn = UIButton()
        btn.backgroundColor = UIColor.red
        btn.titleLabel?.font = UIFont.systemFont(ofSize: 14.0)
        btn.setTitle("收藏", for: .normal)
        btn.setTitleColor(UIColor.white, for: .normal)
        btn.layer.cornerRadius = ADAPTATIONRATIO * 22.0
        btn.layer.masksToBounds = true
        return btn
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        
        self.backgroundColor = UIColor.clear
        self.addSubview(nameLabel)
        self.addSubview(countLabel)
        self.addSubview(tagLabel)
        self.addSubview(personalBtn)
        self.addSubview(collectBtn)
        
        tagLabel.snp.makeConstraints { (make) in
            make.bottom.equalTo(self).offset(-ADAPTATIONRATIO * 26.0)
            make.left.equalTo(self).offset(ADAPTATIONRATIO * 15.0)
        }
        
        countLabel.snp.makeConstraints { (make) in
            make.left.equalTo(self.tagLabel)
            make.bottom.equalTo(self.tagLabel.snp.top).offset(-ADAPTATIONRATIO * 12.0)
        }
        
        nameLabel.snp.makeConstraints { (make) in
            make.left.equalTo(self.tagLabel)
            make.bottom.equalTo(self.countLabel.snp.top).offset(-20)
        }
        
        collectBtn.snp.makeConstraints { (make) in
            make.bottom.equalTo(self.tagLabel.snp.bottom)
            make.right.equalTo(self).offset(-ADAPTATIONRATIO * 20.0)
            make.width.equalTo(ADAPTATIONRATIO * 150.0)
            make.height.equalTo(ADAPTATIONRATIO * 50.0)
        }
        
        personalBtn.snp.makeConstraints { (make) in
            make.bottom.equalTo(self.tagLabel.snp.bottom)
            make.right.equalTo(self.collectBtn.snp.left).offset(-ADAPTATIONRATIO * 20.0)
            make.width.equalTo(ADAPTATIONRATIO * 150.0)
            make.height.equalTo(ADAPTATIONRATIO * 50.0)
        }
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
